nomad/structs/config/plugins.go

```diff
@@ -0,0 +1,75 @@
+package config
+
+import "github.com/mitchellh/copystructure"
+
+// PluginConfig is used to configure a plugin explicitly
+type PluginConfig struct {
```

Comment:
Do we not need a `Type` field here? Or is the type to be inferred from the path in the yet to be written plugin loader?

Reply:
It will be inferred by the plugin loader. The loader will launch the plugins as the base plugin and get the type using the plugininfo RPC